Hestia/app/src/main/java/hestia/backend/NetworkHandler.java

```diff
@@ -4,57 +4,38 @@
 import android.app.Application;
 import com.google.gson.JsonObject;
 import com.google.gson.JsonPrimitive;
-import java.util.ArrayList;
-import java.util.concurrent.CopyOnWriteArrayList;
+
 import hestia.backend.requests.DeleteRequest;
 import hestia.backend.requests.GetDevicesRequest;
 import hestia.backend.requests.GetPluginInformationRequest;
 import hestia.backend.requests.PostRequest;
 
 /**
  * A singleton class which handles interaction between front and back-end. The facade pattern is
- * used to achieve this. During execution, there is a single BackendInteractor accessible
+ * used to achieve this. During execution, there is a single NetworkHandler accessible
  * throughout the entire app through the HestiaApplication class.
  * @see hestia.UI.HestiaApplication
  */
 
-public class BackendInteractor extends Application{
-
-    /**
-     * We use a CopyOnWriteArrayList to avoid ConcurrentModificationExceptions if
-     * a listener attempts to remove itself during event notification.
-     */
-    private final CopyOnWriteArrayList<DevicesChangeListener> listeners =
-            new CopyOnWriteArrayList<>();
-    private static BackendInteractor instance;
-    private ArrayList<Device> devices = new ArrayList<>();
-    private final static String TAG = "BackendInteractor";
-
-    private String ip = "82.73.173.179";
-
-
-    //PI:
-//    private String ip = "82.73.173.179";
-//    private int port = 8022;
-
-    private int port = 8000;
-
+public class NetworkHandler extends Application{
```

Comment:
In that case, we should make all the fields static. It may become uglier this way. Do you have any suggestions?